src/main/webapp/history.js

```diff
@@ -0,0 +1,131 @@
+        `href=${currentUser.loginURL}>login</a> to view your history.`;
+    return;
+  }
+
+  updateUsedSpace(currentUser);
+
+  // Display the photos.
+  const photosResponse = await fetch('/photos');
+  const photos = await photosResponse.json();
+
+  for (const photo of photos) {
+    // This will contain both the image and the delete button.
+    const container = document.createElement('span');
+    container.classList.add('photo-container');
+
+    // Get the image from URL.
+    const imageObj =
+        await getImageFromUrl(`/photo?blob-key=${photo.blobKeyString}`);
+
+    // Convert rectangles returned by the request to Rect objects to be used
+    // by our functions.
+    const responseRects = JSON.parse(photo.jsonBlurRectangles);
+    const blurRects = [];
+    for (const area of responseRects) {
+      let rect;
+      try {
+        rect = new Rect(area, imageObj);
+      } catch (error) {
+        // If rect is invalid, log and ignore it.
+        console.log('Invalid rectangle : ' + error.message,
+            area);
+        continue;
+      }
+
+      blurRects.push(rect);
+    }
+
+    // Blur the image with the default blur radius.
+    const blurRadius = getDefaultBlurRadius(blurRects);
+    const blurredImage = getImageWithBlurredAreas(
+        blurRects, imageObj, blurRadius);
+    blurredImage.classList.add('photo');
+    // Add the image to our container.
+    container.appendChild(blurredImage);
+
+    // Create the delete button and add it to the container.
+    const deleteButton = document.createElement('button');
+    deleteButton.innerHTML = 'DELETE';
+    deleteButton.classList.add('photo-delete');
+    deleteButton.onclick = () => {
+      // Make the button inactive so the user can't click it multiple times.
+      deleteButton.disable = true;
+
+      // Delete the photo from server.
+      fetch(`/photos?photo-id=${photo.id}`, {
+        method: 'DELETE',
+      }).then(async (response) => {
+        if (response.ok) {
+          // Delete the photo from page.
+          contentDiv.removeChild(container);
+
+          // Reload user to update usedSpace.
+          currentUser = await getCurrentUser();
+
+          updateUsedSpace(currentUser);
+        } else {
+          const errorMessage = await response.text();
+          const errorText = response.status + ' server error';
+
+          alert(errorText + ' : ' + errorMessage);
+
+          reject(new Error(errorText));
+          return;
+        }
+      });
+    };
+    container.appendChild(deleteButton);
+
+    // Add the container to the page.
+    contentDiv.appendChild(container);
+  }
+}
+
+/**
+ * Function to display the space used by the user.
+ * @param {User} currentUser
+ */
+function updateUsedSpace(currentUser) {
+  // Convert bytes to megabytes with 2 decimals.
+  const usedSpace = Math.ceil(currentUser.usedSpace / 1024 / 1024 * 100) / 100;
+  document.getElementById('used-space').innerHTML =
+      `Used space: ${usedSpace} / 50 MB`;
```

Comment:
If you change the limit, you'll need to update it here as well. Is there a way to access the variable holding that limit from here?
Alternatively, you can make the limit a member of the user class (the idea of different users having different limit is not unreasonable). That way you can extract it from the currentUser object.

Reply:
There is no way of accessing the variable here other than fetching it from the server.
I created a static variable for the LoggedUser class (so it's the same for all users) and forced gson to include it in the currentUser JSON object.